privacyidea/lib/auditmodules/sqlaudit.py

```diff
@@ -62,6 +64,32 @@
 metadata = MetaData()
 
 
+# Define function to convert SQL DateTime objects to an ISO-format string
+# By using <https://docs.sqlalchemy.org/en/13/core/compiler.html> we can
+# differentiate between different dialects.
+class to_isodate(expression.FunctionElement):
+    name = 'to_isodate'
+
+
+@compiles(to_isodate, 'oracle')
+@compiles(to_isodate, 'postgresql')
+def fn_to_isodate(element, compiler, **kw):
+    return "to_char(%s, 'IYYY-MM-DD HH24:MI:SS')" % compiler.process(element.clauses, **kw)
```

Comment:
I assume for whichever reason the "I" in "IYYY" and the "I" in "MI" is correct?

Reply:
These are the ISO format specifiers for Oracle/Postgres according to the respective docs:
https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/Format-Models.html#GUID-44E8F6D0-7532-4BE1-9300-F9775D9DB027
https://www.postgresql.org/docs/13/functions-formatting.html